### DISCUSSION TITLE
Themes & widgets

### DISCUSSION THREAD

**Discussion by tluyben** [rabbibotton/clog#351] [Ideas]
@rabbibotton You have mentioned themes before in one of the dev journals; while currently there is ultimate freedom by inserting HTML (css) into the code, did you consider a library of widgets (with themes) so people can stick with CL throughout instead of inserting HTML strings etc? One does not exclude what there is now already it might be streamlined a bit to make the default widgets and then add flexibility if you want (in my opinion it would be better to fork the widget library and/or theme (depending what you are doing) and change it). 

Do you have ideas about that direction? 

(I can help with this if there is a plan anyway)

**Comment by rabbibotton** [comment]
To a large degree that is what clog-gui, clog-web and the various high level controls already are. The builder is completely written in clog etc

In fact the builder contains a template for building extensions to clog that also work in the GUI builder and have tutorials on.

I would say build a widget set using that or just following the patterns of CLOG. Take a look at say CLOG-Tree of a non builder extension (although will add).

So I highly encourage you to build widgets as extensions and a few actually already have.

**Reply by rabbibotton** [reply]
Look at how clog gui is theamable and how the builder allows users to change the themes etc

**Reply by tluyben** [reply]
Thanks! I am checking it out!
